ros_cross_compile/ros_cross_compile.py

```diff
@@ -153,17 +188,42 @@ def cross_compile_pipeline(
         default_docker_dir=sysroot_build_context,
         colcon_defaults_file=args.colcon_defaults)
 
+    # stage 1
     if not args.skip_rosdep_collection:
-        gather_rosdeps(
-            docker_client=docker_client,
-            platform=platform,
-            workspace=ros_workspace_dir,
-            skip_rosdep_keys=skip_rosdep_keys,
-            custom_script=custom_rosdep_script,
-            custom_data_dir=custom_data_dir)
-    assert_install_rosdep_script_exists(ros_workspace_dir, platform)
-    create_workspace_sysroot_image(docker_client, platform)
-    run_emulated_docker_build(docker_client, platform, ros_workspace_dir)
+        gather_rosdeps_wrapper((
+            docker_client,
+            platform,
+            ros_workspace_dir,
+            skip_rosdep_keys,
+            custom_rosdep_script,
+            custom_data_dir), dc)
+
+    # get the size of the docker image generated in this stage
+    rosdep_img_size = docker.from_env().images.get('ros_cross_compile:rosdep').attrs['Size']
```

Comment:
There's a lot of implementation-level information leakage happening here. 
Things we shouldn't know or hardcode at this level:
* `"ros_cross_compile:rosdep"` - this is an implementation detail of `gather_rosdeps` and should somehow be coming out of that pipeline step
* `images.get(name).attrs['Size'] / 1000000` - this should be wrapped in the `DockerClient` as a `get_size` API. We shouldn't be calling it here though, it's something that `gather_rosdeps` should be doing for us and adding to the collector within the pipeline stage